### PR TITLE
More pythonic doc for ``bad-chained-comparison``

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

To be real, I wanted to have some fun with this after having the idea in https://discuss.python.org/t/can-we-make-a-syntaxwarning-for-if-x-2-or-3/30263/23?u=pierre-sassoulas, and I also think it's easier to understand than the xor example. Not sure we should remove the xor example though. Took the opportunity to update the generated doc at the same time.

### CODE REVIEW

File: doc/user_guide/configuration/all-options.rst [hunk 1]
```diff
@@ -1285,7 +1285,7 @@ Standard Checkers
 
 --spelling-dict
 """""""""""""""
-*Spelling dictionary name. Available dictionaries: en (aspell), en_AU (aspell), en_CA (aspell), en_GB (aspell), en_US (aspell).*
+*Spelling dictionary name. No available dictionaries : You need to install both the python package and the system dependency for enchant to work..*
```
Comment (DanielNoord) [inline doc/user_guide/configuration/all-options.rst:1288]: Can we remove this change?

Comment (Pierre-Sassoulas) [reply doc/user_guide/configuration/all-options.rst:1288]: I created another commit for this: https://github.com/pylint-dev/pylint/pull/8899/commits/37786d8760cd14fa682aca4f10c57f12ee37378c 
```
 [doc] Upgrade the generated output for python 3.11 without enchant

The default needs to be that you need to install the enchant dict in the doc
```
I conscientiously restored the original value for months to keep it before that, but I think no dict installed make more sense as the default, you're more likely to need the installation info than the info that some dict can be installed.

Comment (Pierre-Sassoulas) [reply doc/user_guide/configuration/all-options.rst:1288]: But I can open a new MR if this is not consensual

**Review by DanielNoord — APPROVED** [review]
I guess it is fine. This also brings us closer to generating the docs for every PR.

**Comment by Pierre-Sassoulas** [comment]
I was waiting to be able to check the result in readthedoc. I must say the framework we put into place to have multiple examples for the same message with directory looks GREAT and is so easy to use.

**Pull request merged**